src/code/mixins/components.ts

```diff
@@ -1,23 +1,36 @@
 import * as React from "react";
 import * as _ from "lodash";
 
-export class Mixin<P, S> extends React.Component<P, S> {
+interface IMixin<P, S> extends React.ComponentLifecycle<P, S> {
```

Comment:
Very nice use of `extends React.ComponentLifecycle<P, S>` to ensure we are future proof from React  API changes.